modules/commons/src/main/java/org/apache/synapse/commons/json/JsonUtil.java

```diff
@@ -468,6 +534,35 @@ private static void removeIndentations(OMElement elem) {
         }
     }
 
+    /**
+     * Remove indentations of a OMElement.
+     * If PRESERVE_SPACES property in the requiredProperties is true, spaces inside XML elements will be preserved.
+     *
+     * @param requiredProperties Required properties
+     * @param elem               OMElement needed to remove indentations
+     */
+    private static void removeIndentations(Map requiredProperties, OMElement elem) {
+        Iterator children = elem.getChildren();
+        while (children.hasNext()) {
+            OMNode child = (OMNode) children.next();
+
+            if (child instanceof OMText) {
+                // preserve spaces of OMText if PRESERVE_SPACES property is true
+                if (requiredProperties.get(Constants.PRESERVE_SPACES) != null
```

Comment:
let's read the PRESERVE_SPACES property from requiredProperties map once to avoid multiple function calls

Reply:
Done